### PR TITLE
Publish placeholders for `Classification` types

### PR DESCRIPTION
**Pull request by benlovell**
Trello: https://trello.com/c/TMbqcbVf

See the commits for the relevant details. I'm fairly green to this area of
whitehall, so feel free to tell me everything I did wrong.

### CODE REVIEW

**Comment by gpeng** [comment]
Looks good 👍

File: db/schema.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@
     t.datetime "created_at"
     t.datetime "updated_at"
     t.string   "title",                    limit: 255
-    t.boolean  "accessible",               limit: 1
```
Comment (jamiecobbett) [inline db/schema.rb]: We shouldn't have these changes. I wonder if maybe you ran the migrations on your host machine (as opposed to your VM) and it has a different MySQL version?

File: test/unit/presenters/publishing_api_presenters/classification_test.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+require 'test_helper'
+
+class PublishingApiPresenters::ClassificationTest < ActiveSupport::TestCase
+  test "presents a valid placeholder 'topic' content item" do
+    topic = create(:topic, name: "Defence and armed forces")
```
Comment (jamiecobbett) [inline test/unit/presenters/publishing_api_presenters/classification_test.rb]: This shouldn't be a `topic`, it should be a `topical_event`.

File: test/unit/presenters/publishing_api_presenters/classification_test.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    public_path = "/government/topics/defence-and-armed-forces"
+
+    expected = {
+      base_path: public_path,
+      publishing_app: "whitehall",
+      rendering_app: "whitehall-frontend",
+      format: "placeholder_topic",
```
Comment (jamiecobbett) [inline test/unit/presenters/publishing_api_presenters/classification_test.rb]: I think we should use a `format` of `policy_area`/`placeholder_policy_area` to be consistent with the label we use on [GOV.UK](https://www.gov.uk/government/topics/public-health). After talking with @rboulton, we also settled on `policy_areas` for the key in the links hash which will contain these things. See: https://github.com/alphagov/govuk-content-schemas/blob/master/formats/statistics_announcement/publisher/links.json#L13

File: db/schema.rb (hunk 1 above)
Comment (benlovell) [inline db/schema.rb]: It is explained in the commit message, with a link to the rails change/issue.

File: db/data_migration/20160202115027_publish_classification_placeholders.rb [hunk 4]
```diff
@@ -0,0 +1 @@
+Classification.where.not(content_id: nil).each(&:publish_to_publishing_api)
```
Comment (benilovj) [inline db/data_migration/20160202115027_publish_classification_placeholders.rb]: FYI, there's a [rake task that does this](https://github.com/alphagov/whitehall/blob/master/lib/tasks/publishing_api.rake#L42), so we could potentially skip this altogether...

File: db/data_migration/20160202102043_generate_content_ids_on_classifications.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+require 'securerandom'
+
+Classification.where(content_id: nil).each do |classification|
+  classification.update_column(:content_id, SecureRandom.uuid)
```
Comment (benilovj) [inline db/data_migration/20160202102043_generate_content_ids_on_classifications.rb]: FYI, there's a [rake task that does this](https://github.com/alphagov/whitehall/blob/master/lib/tasks/publishing_api.rake#L42), so we could potentially skip this altogether...

File: db/schema.rb (hunk 1 above)
Comment (benilovj) [inline db/schema.rb]: I'd prefer to have the incidental changes on a separate PR...

File: db/data_migration/20160202102043_generate_content_ids_on_classifications.rb (hunk 5 above)
Comment (benilovj) [inline db/data_migration/20160202102043_generate_content_ids_on_classifications.rb]: After a conversation in person, the consensus is that data migrations are a better way to do this than a rake task...

File: test/unit/presenters/publishing_api_presenters/classification_test.rb (hunk 3 above)
Comment (jamiecobbett) [inline test/unit/presenters/publishing_api_presenters/classification_test.rb]: There's a precedent for this with WorkingGroups: https://github.com/alphagov/whitehall/blob/master/app/presenters/publishing_api_presenters/working_group.rb

Comment (benlovell) [inline test/unit/presenters/publishing_api_presenters/classification_test.rb]: @jamiecobbett Ah, good spot. Another reason why the partial rename of `Topic` / `PolicyArea` was a Bad Thing :tm: I'll see to this.

**Comment by benlovell** [comment]
@jamiecobbett Should be good to go now.

**Comment by jamiecobbett** [comment]
Looks good, merge when you're ready :+1:

**Pull request merged**